### PR TITLE
feat(`publick8s`): add an ARM cluster node pool

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3584
create an ARM pool based on [Standard_D8ps_v5](https://learn.microsoft.com/en-us/azure/virtual-machines/dv5-dsv5-series#dsv5-series)

### CODE REVIEW

**Comment by smerle33** [comment]
Looks good, but need to make sure about the `os_disk_size_gb` as there is no temp storage on D8ps_v5 https://learn.microsoft.com/en-us/azure/virtual-machines/dpsv5-dpdsv5-series but a OS disk of 1023 GiB seems present : https://azureprice.net/vm/Standard_D8ps_v5

```
Terraform will perform the following actions:

  # azurerm_kubernetes_cluster_node_pool.publicarmpool will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "publicarmpool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = "..."
      + max_count             = 10
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "pubarmpool"
      + node_count            = (known after apply)
      + node_labels           = (known after apply)
      + orchestrator_version  = (known after apply)
      + os_disk_size_gb       = (known after apply)
      + os_disk_type          = "Ephemeral"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + scale_down_mode       = "Delete"
      + spot_max_price        = -1
      + tags                  = {
          + "repository" = "jenkins-infra/azure"
          + "scope"      = "terraform-managed"
        }
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_D8ps_v5"
      + vnet_subnet_id        = "..."
      + zones                 = [
          + "3",
        ]
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

File: publick8s.tf [hunk 1]
```diff
@@ -81,6 +81,20 @@ resource "azurerm_kubernetes_cluster_node_pool" "publicpool" {
   tags                  = local.default_tags
 }
 
+resource "azurerm_kubernetes_cluster_node_pool" "publicarmpool" {
+  name                  = "pubarmpool"
+  vm_size               = "Standard_D8ps_v5" # 8 vCPU, 32 GB RAM, 0 GB disk, 12 800 IOPS
+  os_disk_type          = "Ephemeral"
+  os_disk_size_gb       = 200 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dv5-dsv5-series#dsv5-series (depends on the instance size)
```
[the remark this answers is not in the document]
Comment (dduportal) [reply publick8s.tf:88]:
```suggestion
  os_disk_size_gb       = 200 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dpsv5-dpdsv5-series#dpdsv5-series (depends on the instance size)
```

**Comment by smerle33** [comment]
size needed for JAVADOC :

**Review by dduportal — DISMISSED** [review]
YOu can set it ready for review and merge whenever you are ready!

File: publick8s.tf [hunk 2]
```diff
@@ -81,6 +81,34 @@ resource "azurerm_kubernetes_cluster_node_pool" "publicpool" {
   tags                  = local.default_tags
 }
 
+resource "azurerm_kubernetes_cluster_node_pool" "x86medimum" {
```
Comment (dduportal) [inline publick8s.tf:84]:
```suggestion
resource "azurerm_kubernetes_cluster_node_pool" "x86medium" {
```

File: publick8s.tf [hunk 3]
```diff
@@ -81,6 +81,34 @@ resource "azurerm_kubernetes_cluster_node_pool" "publicpool" {
...
+  name                  = "x86medimum"
```
Comment (dduportal) [inline publick8s.tf:85]:
```suggestion
  name                  = "x86medium"
```

**Review by dduportal — CHANGES_REQUESTED** [review]
Can you explain why we decided to add the new x86 medium (and explain the next step) please?

**Comment by smerle33** [comment]
> Can you explain why we decided to add the new x86 medium (and explain the next step) please?

as this PR create a dedicated ARM pool with a medium size machine, we decided to homogenize the other pool name (publicpool) in order to be able to provide others kind of node pools in the future (big for example).

**Comment by smerle33** [comment]
here is the steps we choose : 

- micro 2cpu/8G
- small 4cpu/16G
- medium 8cpu/32G
- big 16cpu/64G

**Review by dduportal — APPROVED** [review]
`Plan: 2 to add, 0 to change, 0 to destroy.` 

LGTM

**Pull request merged**